my-webxr-app/tests/Axis.test.tsx

```diff
@@ -34,6 +34,9 @@ describe('Axis Tests', () => {
     const renderer = await ReactThreeTestRenderer.create(<Element />);
     // wait for scene children to be rendered
     await waitFor(() => expect(renderer.scene.children).toBeDefined());
+    await waitFor(() => expect(renderer.scene.children !== null).toBeTruthy());
+    await waitFor(() => expect(renderer.scene.children[1]).toBeDefined());
+    await waitFor(() => expect(renderer.scene.children[1] !== null).toBeTruthy());
```

Comment:
So this issue was present before your changes? Is so, that's weird how it was missed/permitted by the CI/CD test workflow.

Reply:
yes, I was running it on just ID5 base branch and it was happening. It has something to do with the axis and taking too long to render. This did not entirely fix the issue but it made it more predictable.

Reply:
Alright. I'll take a look into this more tonight.

Reply:
I think I see what's happening here. So, for me at least, the test only fails sometimes—depending on whether or not the children have loaded yet or not. It's a bit of a race condition it seems like. The test failed on 4/15 runs for me.
> ![Screenshot 2024-03-25 at 18 35 04](https://github.com/UniversityOfSaskatchewanCMPT371/term-project-2024-team-2/assets/76539691/5c119a12-66e6-47a0-8f18-6049cbe8c9ed)
>  220-File-Name-Fix

It seems like these `waitFor()` checks are on the right track, but not entirely there. Between this branch and ID5, the test fails on line 38's expect statement vs an undefined reference on line 39.
> ![Screenshot 2024-03-25 at 18 41 31](https://github.com/UniversityOfSaskatchewanCMPT371/term-project-2024-team-2/assets/76539691/ef1ba559-85b2-49c1-a83e-c7945c24f2b0)
> release/ID5

Reply:
Yes that is what me and trang were thinking, and maybe the process for running the tests on the github is just slower then on our computers, giving it more time to render/load